analysis.md

```diff
@@ -0,0 +1,77 @@
+
+### Admins: Creation of a New WMI Object
+If the user has administrator permissions, the script creates a permanent WMI event subscription that watches for a user logon, waits 30 minutes, then executes the layer 3 payload. At this point, the program has exhibited enough strange behavior for the SentinelAgent to identify it as malware.
+
+### Scheduled Task
+The script also creates a scheduled task to run the malware 30 minutes after a user logs on. The method used to start the malware differs only slightly depending on if the user has administrator permissions. If the user is a non-admin, `wscript` is used to execute the contents of `kernel32.vbs`. Otherwise, a Invoke-Expression (IEX) cmdlet is used.
+
+## Layer 3
+Now that we've discussed the various ways the attacker achieves persistence of the malware, lets find out what the layer 3 payload is responsible for. You'll notice yet another Base64 blob inside layer 2.
+
+You'll notice an attempt to thwart static analysis at the creation of a mutex inside the function `logic`, named `"SourceFireSux"`, a jab aimed at the security company. It may seem that the function exits prompty, but the boolean variable is set to `True` after the [construction of the mutex](https://msdn.microsoft.com/en-us/library/bwe34f1k(v=vs.110).aspx).
+
+```
+[bool]$var = $false;
+$mutex = New-Object System.Threading.Mutex($true, “SourceFireSux”, [ref] $var);        
+if (!$var)
+{
+  exit;
+}
+```
+
+After decoding and deobfuscating, it is apparent that the code is repeatedly querying for DNS text records. Responses from these queries dictate behavior of the program: whether it should sleep, continue, or stop running. A result of `"idle"` would cause the process to sleep between 3500 and 5400 seconds before continuing. A result of `"stop"` would prompt the process to exit.
+
+I suspect that the purpose of these text record queries is to retrieve yet another payload layer. Unfortunately, since these domains are not active, there is no way for us to go further. However, analysis on this script has been done [here](https://blog.talosintelligence.com/2017/03/dnsmessenger.html).\
+
+## Layer 4
+Thanks to Talos, we know what the fourth layer playload looks like.
+
+![image](https://2.bp.blogspot.com/-S3hr64DjZCE/WLhNpISvZQI/AAAAAAAAAPA/QyjkYq1h83IpdAXhEkjDhb22s05h7LGWQCLcB/s1600/image15.png)
+
+The `dec` function in layer 3 is used to decode this Base64 string and the result is passed to the Invoke-Expression cmdlet. According to Talos's analysis, the script redirects STDIN, STDOUT, and STDERR so the attacker can read from and write to the command line processor. The payload performs more DNS lookups and establishes a communication channel with a command and control server. From here, the attacker may send commands to be executed on the victim machine's command line interpreter and receive the results of those commands, all through DNS txt queries and responses.
```

Comment:
This needs a summary, something like "we just walked through an older but interesting malware. i learned some stuff. i hope you learned some stuff. thanks for reading!"